web3/utils/validation.py

```diff
@@ -23,8 +23,8 @@ def validate_address(address):
     Helper function for validating an address
     """
     if not is_address(address):
+        validate_address_checksum(address)
```

Comment:
This seems weird.  What am I missing.  This is now inside of the block of code that should only be reachable if it is not an address..  Why would this be here?